src/IO/AzureBlobStorage/copyAzureBlobStorageFile.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "config.h"
+
+#if USE_AZURE_BLOB_STORAGE
+
+#include <Storages/StorageAzureBlobCluster.h>
+#include <Storages/StorageAzureBlob.h>
+#include <Interpreters/threadPoolCallbackRunner.h>
+#include <base/types.h>
+#include <functional>
+#include <memory>
+
+
+namespace DB
+{
+class SeekableReadBuffer;
+
+using CreateReadBuffer = std::function<std::unique_ptr<SeekableReadBuffer>()>;
+
+/// Copies a file from AzureBlobStorage to AzureBlobStorage.
+/// The parameters `src_offset` and `src_size` specify a part in the source to copy.
+void copyAzureBlobStorageFile(
+    MultiVersion<Azure::Storage::Blobs::BlobContainerClient> & src_client,
+    MultiVersion<Azure::Storage::Blobs::BlobContainerClient> & dest_client,
+    const String & src_container,
+    const String & src_blob,
+    size_t src_offset,
+    size_t src_size,
+    const String & dest_container,
+    const String & dest_blob,
+    std::shared_ptr<AzureObjectStorageSettings> settings,
+    const ReadSettings & read_settings,
+    const std::optional<std::map<String, String>> & object_metadata = std::nullopt,
```

Comment:
How do you use `object_metadata` in this function?

Reply:
not used, will remove it

Reply:
addressed it